### PR TITLE
BUGZ 1029: changing raw pointer to unique pointer with custom deleter

### PR DESCRIPTION
**Pull request by amerhifi**
follow up to 1029 bugz. moving to using unique_ptr with custom deleter.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20606/HighFidelity-Beta-PR15995-087b19d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20606/HighFidelity-Beta-Interface-PR15995-087b19d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20608/HighFidelity-Beta-PR15995-de0b038.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20608/HighFidelity-Beta-Interface-PR15995-de0b038.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20608/HighFidelity-Beta-PR15995-de0b038.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20608/HighFidelity-Quest-Beta-PR15995-de0b038.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20608/HighFidelity-Beta-PR15995-de0b038.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20608/HighFidelity-Beta-Interface-PR15995-de0b038.exe

File: interface/src/ui/InteractiveWindow.cpp [hunk 1]
```diff
@@ -136,12 +136,13 @@ InteractiveWindow::InteractiveWindow(const QString& sourceUrl, const QVariantMap
     if (properties.contains(PRESENTATION_MODE_PROPERTY)) {
         presentationMode = (InteractiveWindowPresentationMode) properties[PRESENTATION_MODE_PROPERTY].toInt();
     }
+    
+   _interactiveWindowProxy= std::unique_ptr<InteractiveWindowProxy, std::function<void(InteractiveWindowProxy*)>>(new InteractiveWindowProxy, [](InteractiveWindowProxy *p) {
```
Comment (huffman) [inline interface/src/ui/InteractiveWindow.cpp]: This looks great 👍 

To make it a little easier to read, could we split it up? The style guide suggests lines be less than 128 characters.

File: interface/src/ui/InteractiveWindow.cpp [hunk 2]
```diff
@@ -136,12 +136,13 @@ InteractiveWindow::InteractiveWindow(const QString& sourceUrl, const QVariantMap
...
+       p->deleteLater();
+   });
 
-    if (!_interactiveWindowProxy) {
-        _interactiveWindowProxy = new InteractiveWindowProxy();
-        QObject::connect(_interactiveWindowProxy, &InteractiveWindowProxy::webEventReceived, this, &InteractiveWindow::emitWebEvent, Qt::QueuedConnection);
-        QObject::connect(this, &InteractiveWindow::scriptEventReceived, _interactiveWindowProxy, &InteractiveWindowProxy::emitScriptEvent, Qt::QueuedConnection);
-    }
+    QObject::connect(_interactiveWindowProxy.get(), &InteractiveWindowProxy::webEventReceived, this, &InteractiveWindow::emitWebEvent, Qt::QueuedConnection);
+    QObject::connect(this, &InteractiveWindow::scriptEventReceived, _interactiveWindowProxy.get(), &InteractiveWindowProxy::emitScriptEvent, Qt::QueuedConnection);
```
Comment (huffman) [inline interface/src/ui/InteractiveWindow.cpp]: These also look fairly long - the `QObject::` can actually be removed here, which might be enough.

File: interface/src/ui/InteractiveWindow.h [hunk 3]
```diff
@@ -41,7 +41,7 @@ class InteractiveWindowProxy : public QObject {
     Q_OBJECT
 public:
     InteractiveWindowProxy(){}
-
+
```
Comment (huffman) [inline interface/src/ui/InteractiveWindow.h]: Was this whitespace accidentally added?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20624/HighFidelity-Beta-PR15995-b3ea9d0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20624/HighFidelity-Beta-Interface-PR15995-b3ea9d0.dmg

File: interface/src/ui/InteractiveWindow.cpp [hunk 4]
```diff
@@ -136,12 +136,16 @@ InteractiveWindow::InteractiveWindow(const QString& sourceUrl, const QVariantMap
     if (properties.contains(PRESENTATION_MODE_PROPERTY)) {
         presentationMode = (InteractiveWindowPresentationMode) properties[PRESENTATION_MODE_PROPERTY].toInt();
     }
+    
+   _interactiveWindowProxy= std::unique_ptr<InteractiveWindowProxy,
```
Comment (huffman) [inline interface/src/ui/InteractiveWindow.cpp]: Minor style issue: there should be a space before the `=`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20626/HighFidelity-Beta-PR15995-85fa510.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20626/HighFidelity-Beta-Interface-PR15995-85fa510.dmg

**Comment by amerhifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20626/HighFidelity-Beta-PR15995-85fa510.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20626/HighFidelity-Quest-Beta-PR15995-85fa510.apk)

**Comment by amerhifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20633/HighFidelity-Beta-PR15995-85fa510.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20633/HighFidelity-Beta-Interface-PR15995-85fa510.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20633/HighFidelity-Beta-PR15995-85fa510.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20633/HighFidelity-Quest-Beta-PR15995-85fa510.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20633/HighFidelity-Beta-PR15995-85fa510.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20633/HighFidelity-Beta-Interface-PR15995-85fa510.dmg

**Pull request merged**